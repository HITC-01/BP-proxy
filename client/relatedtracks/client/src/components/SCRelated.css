ul, li {
  padding: 0;
  list-style: none;
}

body {
  font-family: 'Overpass', sans-serif;
  width: 300px;
  padding: 5px;
  margin: 0;
}

.songContainer {
  position: relative;
  font-size: 14px;
  border-bottom: 0.5px solid #999;
}

.songContainerHeader {
  font-weight: 100;
  margin: 0;
  display: inline-block;
  color: #999;
}

.rel_actualTitle {
  font-weight: normal;
  font-size: 14px;
  color: #999; 
}

.viewAll {
  color: #999;
  float: right;
}

.rel_hover:hover {
  color: #000;
}
