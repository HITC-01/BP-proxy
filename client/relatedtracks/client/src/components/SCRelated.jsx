import React from 'react';
import PropTypes from 'prop-types';
import AddTrack from './AddTrack';
import styles from './SCRelated.css';

class SCRelated extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      songs: [],
    };
    this.fetchRelatedSongs = this.fetchRelatedSongs.bind(this);
  }

  componentDidMount() {
    this.fetchRelatedSongs()
      .then((songs) => {
        this.setState({ songs });
      })
      .catch(() => this.setState({ songs: [] }));
  }

  fetchRelatedSongs() {
    const { url } = this.props;
    // console.log('test');
    return fetch(`${url}/related/api/songs/${this.songId}/related`, {
      headers: {
        'Content-Type': 'application/json',
      },
      credentials: 'same-origin',
    })
      .then(function (response) {
        return response.json()
      })
      .then((responseAsJson) => {
        const list = JSON.parse(responseAsJson[0].related_songs);
        return Promise.all(list.map(song => {
          let url = `${url}/related/api/songs/${song}/`
          return fetch(url, {
            headers: {
              "Content-Type": "application/json"
            },
            credentials: "same-origin"
          })
            .then(function (response) {
              return response.json()
            })
            .then(responseSong => {
              return responseSong[0];
            })
            .catch(err => console.log(err));
        }))
          .then(songs => {
            return songs;
          })
      })
      .catch(err => console.log(err));
  }

  render() {
    const { songs } = this.state;
    return (
      <div>
        <div className={styles.songContainer}>
          <h3 className={styles.songContainerHeader}>
            <span>
              <i className="fab fa-soundcloud" />
            </span>
            <span className={styles.rel_actualTitle}>Related tracks</span>
          </h3>
          <span className={`${styles.viewAll} ${styles.rel_hover}`}>View all</span>
        </div>
        <ul>
          <AddTrack onLoad={songs} />
        </ul>
      </div>
    );
  }
}
SCRelated.propTypes = {
  url: PropTypes.string,

};

SCRelated.defaultProps = {
  url: '',
};

export default SCRelated;
