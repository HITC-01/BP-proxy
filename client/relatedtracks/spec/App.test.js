import React from 'react';
import renderer from 'react-test-renderer';
import 'isomorphic-fetch';
import SCRelated from '../client/src/components/SCRelated';

describe('<SCRelated />', () => {
  test('Should render correctly', () => {
    const component = renderer.create(<SCRelated url="http://localhost:3002" />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
