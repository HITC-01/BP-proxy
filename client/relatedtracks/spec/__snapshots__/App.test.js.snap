// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SCRelated /> Should render correctly 1`] = `
<div
  className="songDataPlaceholder"
>
  <div
    className="songContainer"
  >
    <h3
      className="songContainerHeader"
    >
      <span>
        <i
          className="fab fa-soundcloud"
        />
      </span>
      <span
        className="rel_actualTitle"
      >
        Related tracks
      </span>
    </h3>
    <span
      className="viewAll rel_hover"
    >
      View all
    </span>
  </div>
  <ul
    className="list-items"
  />
</div>
`;
